quarkus/config-api/src/main/java/org/keycloak/config/HttpOptions.java

```diff
@@ -110,4 +110,11 @@ public enum ClientAuth {
             .description("Enables or disables the HTTP/s and Socket serving.")
             .defaultValue(Boolean.TRUE)
             .build();
+
+    public static final Option<Integer> HTTP_MAX_CONCURRENT_REQUESTS = new OptionBuilder<>("http-max-concurrent-requests", Integer.class)
```

Comment:
I wonder if we s hould make it build time for now so we could completely avoid creating the request filter if not configured. It might make sense to make this fully opt-in as it's only experimental.

On the other hand, the current logic in the handler is effectively skipping all the logic, so we should be safe as is.

Reply:
Enabling and disabling it could be a build-time option. But as this option takes a number as a parameter of how many requests to handle in parallel, I argue it should be a runtime parameter. 

To make this a built-time option, we would need IMHO a second option (maybe a feature flag). I remember we discussed it previously for the health probe ticket and didn't want to use that option there. 

Looking at the code, it should be fast enough as it is checking for null if it is not set. 

So I'd suggest to proceed as is, also as this is a preview feature and we want to get feedback before we finalize it.